### PR TITLE
[stable/traefik] Generate self sign default cert

### PR DESCRIPTION
**Pull request by kozmagabor**
Signed-off-by: Gabor Kozma <gabo@kozmagabor.com>

#### What this PR does / why we need it:
I want to generate cert with specific information. (CN,SAN etc.)

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
Cloud you check it  @dtomcej  @emilevauge @krancour @ldez, please?

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/traefik/templates/default-cert-secret.yaml [hunk 1]
```diff
@@ -10,6 +10,14 @@ metadata:
     heritage: {{ .Release.Service | quote }}
 type: Opaque
 data:
+{{- if .Values.ssl.generateTLS }}
+  {{- $ca := genCA "default-ca" 365 }}
+  {{- $cn := .Values.ssl.defaultCN }}
```
Comment (dtomcej) [inline stable/traefik/templates/default-cert-secret.yaml]: Default CN should be optional.

File: stable/traefik/values.yaml [hunk 2]
```diff
@@ -65,6 +65,14 @@ ssl:
   permanentRedirect: false
   upstream: false
   insecureSkipVerify: false
+  generateTLS: false
+  defaultCN: ""
```
Comment (dtomcej) [inline stable/traefik/values.yaml]: should be commented out by default.

**Review by dtomcej — CHANGES_REQUESTED** [review]

**Comment by dtomcej** [comment]
I like the sprig-foo :) Just some comments for usability :)

File: stable/traefik/templates/default-cert-secret.yaml (hunk 1 above)
Comment (dtomcej) [reply stable/traefik/templates/default-cert-secret.yaml]: All values but `generateTLS` should be optional (aka need a test for existence before use)

**Comment by kozmagabor** [comment]
Thanks, @dtomcej I updated my PR

**Comment by dtomcej** [comment]
@kozmagabor Can you bump the chart version?

**Comment by kozmagabor** [comment]
@dtomcej It's already bumped to 1.53.0

**Comment by kozmagabor** [comment]
Ohh, sorry. You are right!  I fixed it.

**Comment by kozmagabor** [comment]
@dtomcej Could you check it again? Thanks!

**Comment by kozmagabor** [comment]
@juliens @nmengin Could you check it?

**Comment by dtomcej** [comment]
Hello @kozmagabor, can you bump the version, so that we can get it merged? Sorry for the delay, catching up after Kubecon.

**Comment by kozmagabor** [comment]
Hi @dtomcej 

Could you check it again?

**Comment by kozmagabor** [comment]
Anybody else? @juliens @nmengin

**Pull request merged**